rule-engine/README.md

```diff
@@ -0,0 +1,6 @@
+# Rule Engine 
+
+This project implements an engine to perform real-time security rule matching against a stream of incoming data.
```

Comment:
Can you specify how this engine is going to be used in the pipelines?

Reply:
This project will be semi-separate from the rest of DataPrepper, but it will be integrated into DataPrepper through a processor that leverages the rule engine.

The long term vision is to perform security rule analysis against data moving through the pipeline with this rule engine

Reply:
What do you mean by semi-separate?

It seems this may be a good candidate for a new project in the opensearchproject. You could make a library publish to Maven Central and then use that in Data Prepper. This would be especially helpful if you want a different group of maintainers to review changes to that core library. 

Can you elaborate some on the vision here to see if this is the right project to use?